<a href="https://colab.research.google.com/github/Beagan19/Future-projects/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
## Pandas
import pandas as pd


## Load Dataset 1

In [3]:
metabolic_df = pd.read_csv('/content/drive/MyDrive/Project 2- ML/Metabolic Syndrome.csv')

In [4]:
metabolic_df.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [5]:
metabolic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2401 entries, 0 to 2400
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   seqn               2401 non-null   int64  
 1   Age                2401 non-null   int64  
 2   Sex                2401 non-null   object 
 3   Marital            2193 non-null   object 
 4   Income             2284 non-null   float64
 5   Race               2401 non-null   object 
 6   WaistCirc          2316 non-null   float64
 7   BMI                2375 non-null   float64
 8   Albuminuria        2401 non-null   int64  
 9   UrAlbCr            2401 non-null   float64
 10  UricAcid           2401 non-null   float64
 11  BloodGlucose       2401 non-null   int64  
 12  HDL                2401 non-null   int64  
 13  Triglycerides      2401 non-null   int64  
 14  MetabolicSyndrome  2401 non-null   object 
dtypes: float64(5), int64(6), object(4)
memory usage: 281.5+ KB


1. Source of data
- https://data.world/informatics-edu/metabolic-syndrome-prediction

2. Brief description of data
- This dataset uses multiple columns that represent different common risk factors to predict whether a person has metabolic syndrome or not.

In [6]:
metabolic_df.shape

(2401, 15)

In [7]:
metabolic_df.duplicated().sum()

0

In [8]:
metabolic_df.isna().sum()

seqn                   0
Age                    0
Sex                    0
Marital              208
Income               117
Race                   0
WaistCirc             85
BMI                   26
Albuminuria            0
UrAlbCr                0
UricAcid               0
BloodGlucose           0
HDL                    0
Triglycerides          0
MetabolicSyndrome      0
dtype: int64

In [9]:
## Define features (X) and target (y)
target = 'MetabolicSyndrome'
labels=['No Metabolic Syndrome', 'Metabolic Syndrome']

X = metabolic_df.drop(columns = [target]).copy()
y = metabolic_df[target].copy()

3. What is the target?
- 'MetabolicSyndrome' which states whether a person does or does not have metabolic syndrome

4. What does one row represent? (A person?  A business?  An event? A product?)
- One row represents a person.

5. Is this a classification or regression problem?
- This is a classification problem.

In [10]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2401 entries, 0 to 2400
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   seqn           2401 non-null   int64  
 1   Age            2401 non-null   int64  
 2   Sex            2401 non-null   object 
 3   Marital        2193 non-null   object 
 4   Income         2284 non-null   float64
 5   Race           2401 non-null   object 
 6   WaistCirc      2316 non-null   float64
 7   BMI            2375 non-null   float64
 8   Albuminuria    2401 non-null   int64  
 9   UrAlbCr        2401 non-null   float64
 10  UricAcid       2401 non-null   float64
 11  BloodGlucose   2401 non-null   int64  
 12  HDL            2401 non-null   int64  
 13  Triglycerides  2401 non-null   int64  
dtypes: float64(5), int64(6), object(3)
memory usage: 262.7+ KB


6. How many features does the data have?
- There are 14 features.


7. How many rows are in the dataset?
- There are 2401 row.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
- missing data
- unbalanced data

## Load Dataset 2

In [11]:
wine_df = pd.read_csv('/content/drive/MyDrive/Project 2- ML/wines_SPA.csv')

In [12]:
wine_df.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

1. Source of data
- https://data.world/informatics-edu/metabolic-syndrome-prediction

2. Brief description of data
- The dataset uses multiple columns to represent a bottle of spanish wine. The task is to predict prices using the given data.

In [13]:
wine_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       7500 non-null   object 
 1   wine         7500 non-null   object 
 2   year         7498 non-null   object 
 3   rating       7500 non-null   float64
 4   num_reviews  7500 non-null   int64  
 5   country      7500 non-null   object 
 6   region       7500 non-null   object 
 7   price        7500 non-null   float64
 8   type         6955 non-null   object 
 9   body         6331 non-null   float64
 10  acidity      6331 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 644.7+ KB


In [14]:
wine_df.shape

(7500, 11)

In [15]:
wine_df.duplicated().sum()

5452

In [16]:
# drop duplicate rows
wine_df = wine_df.drop_duplicates()
wine_df.duplicated().sum()

0

In [17]:
wine_df.isna().sum()

winery           0
wine             0
year             2
rating           0
num_reviews      0
country          0
region           0
price            0
type           106
body           271
acidity        271
dtype: int64

In [18]:
wine_df.isna().sum()

winery           0
wine             0
year             2
rating           0
num_reviews      0
country          0
region           0
price            0
type           106
body           271
acidity        271
dtype: int64

In [19]:
## Define features (X) and target (y)
target = 'price'

X = wine_df.drop(columns = [target]).copy()
y = wine_df[target].copy()

3. What is the target?
- 'price' which states how much a single bottle costs in euros.

4. What does one row represent? (A person?  A business?  An event? A product?)
- One row represents a bottle of wine.

5. Is this a classification or regression problem?
- This is a regression problem.

In [20]:
X.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2048 entries, 0 to 6100
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       2048 non-null   object 
 1   wine         2048 non-null   object 
 2   year         2046 non-null   object 
 3   rating       2048 non-null   float64
 4   num_reviews  2048 non-null   int64  
 5   country      2048 non-null   object 
 6   region       2048 non-null   object 
 7   type         1942 non-null   object 
 8   body         1777 non-null   float64
 9   acidity      1777 non-null   float64
dtypes: float64(3), int64(1), object(6)
memory usage: 176.0+ KB


6. How many features does the data have?
- There are 10 features.

7. How many rows are in the dataset?
- There are 2048 rows.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
- missing data
